### DISCUSSION TITLE
Empfehlung Wetterprognose-Anbieter

### DISCUSSION THREAD

**Discussion by jaydee73** [wiggal/GEN24_Ladesteuerung#59] [General]
Hi wiggal,

es sind ja mittlerweile drei Anbieter verfügbar bei dir. Bisher arbeite ich mit solarprognose. Ist ja ein deutscher Anbieter und ich habe bisher vermutet (reines Bauchgefühl, unvalidiert...), dass die Daten hier die höchste Genauigkeit haben. Allerdings stelle ich fest, dass das doch häufiger mal nicht passt... ;-)

Hast du Empfehlungen, welcher Anbieter die beste Datenbasis hat? Oder kann man das überhaupt sagen? Oder anders gefragt: Was nutzt du? Oder offen gefragt in die Runde: Was nutzt ihr so?

Gruß,
JD

**Comment by wiggal** [comment]
Servus,
ich verwende aktuell https://toolkit.solcast.com.au und bin der Meinung, dass dies die genauesten Prognosen liefert.
Aber vielleicht kann sich dazu noch jemand äußern, der zu fällig hier mit liest.

Gruß
Wiggal

**Comment by jaydee73** [comment]
Ok, Danke. Ich verstehe das richtig, dass mit der neuesten Version 0.13.2 nun auch zwei Dächer (z. B. Ost-West-Ausrichtung) über solcast abgerufen werden können und von deinem Script dann in eine WeatherData.json zusammengefasst werden?

**Reply by wiggal** [reply]
ja genau, du muss auf der solcast Seite zwei Ausrichtungen konfigurieren und kannst sie dann in det config.ini konfigurieren. Dann werden die beiden Ausrichtungen zusammengefasst. Ich hab den Fall nicht, aber @zellneralex hat dies rein programmiert. siehe letzte Diskussion.
https://github.com/wiggal/GEN24_Ladesteuerung/discussions/55

**Reply by jaydee73** [reply]
Ok, Danke. Ich überlege, wie ich am einfachsten mal einige Tage beide Dienste abfrage und sozusagen vergleiche. Aber beide schreiben ja in die gleiche weatherData.json. Fällt dir da ein charmanter Weg ein, um das irgendwie parallel laufen zu lassen (läuft bei mir als Docker-Container)?

**Reply by wiggal** [reply]
Welchen Wetterdienst willst du nur testweise abfragen,
dann bau ich dir schnell ein Image in dem der zweite Wetterdienst in eine weatherData2.json schreibt
Müssen wir halt überlegen wie ich dir das Image übermittle

**Reply by zellneralex** [reply]
Brauchst nix extra bauen, leg ein 2tes file an und ändere den Namen in der config.ini

so hab ich es auch gemacht als ich auf 2 Sites umgebaut und getestet habe. In jeden config Block mussein file angegeben werden. Das muss also nicht das gleiche sein, es muss aber existieren damit es geht.

**Reply by wiggal** [reply]
ja genau, ich kenn mich in meinem eigenen Prog nicht aus. 🤣
einfach den Namen in der Variable "weatherfile" im Block des zu testenden Wetterdienst ändern.

**Reply by jaydee73** [reply]
Ja das ist ja cool. Danke für den Tipp. Dann versuche ich das mal die Tage und werde einige Tage lang die Werte parallel beobachten (und am Fronius jeweils schauen, was wirklich reingekommen ist pro Tag).

**Reply by jaydee73** [reply]
hmmm...irgendwas fluppt da noch nicht richtig. Ich habe Solcast aufgesetzt und gemäß Wiki folgenden Crontab-Eintrag hinzugefügt:

1 6,8,11,13,15 * * * /home/GEN24/start_PythonScript.sh Solcast_WeatherData.py

Damit sollte nach meinem Verständnis das Script um 6.01, 8.01, 11.01, 13.01 und 15.01 Uhr laufen.

Gelaufen ist es aber anscheinend nur um 11.01 Uhr. Die dazugehörige weatherData2.json hat einen Timestamp von 11.01 Uhr und die Solcast-API sagt auch "heute 2 Abrufe" (was ja passt, pro angegebener Zeit sind es ja zwei Abrufe für West- und Ostdach).

Aber grundsätzlich funktioniert es ja anscheinend, sonst wäre es um 11.01 Uhr ja nicht einmalig durchgelaufen. Die weatherData2.json wurde auch mit plausiblen Werten gefüllt. 

Hast du vielleicht ne Idee, warum es zu den weiteren Uhrzeiten nicht geklappt hat?

**Reply by wiggal** [reply]
Wenn es aus irgend welchen Gründen (evtl. dataAgeMaxInMinutes = 1000) müsste es zum jeweiligen Zeitpunkt (6.01, 8.01 usw.)
in der Crontab.log stehen.
Wenn du nur zwei Abrufe hast, hast du vermutlich no_history = 1 gesetzt, oder?

**Reply by jaydee73** [reply]
Ja, natürlich. Ich Depp....dataAgeMax steht defaultmäßig auf 1000. Hatte ich nicht geändert. Ich mach mal 100 draus.

Ja, no_history hatte ich auf 1 gesetzt, damit ich für beide Sites zumindest 5 Abfragen pro Tag machen darf.

**Comment by jaydee73** [comment]
Sodele...ich habe mal ne gute Woche lang beide Anbieter erfasst und mit dem realen Ertrag verglichen. Falls es jemanden interessiert, hier meine Ergebnisse:

<img width="307" alt="Bildschirmfoto 2024-04-20 um 09 13 20" src="https://github.com/wiggal/GEN24_Ladesteuerung/assets/21342601/906a9555-a6d0-4640-9757-b53d2a110e6f">

In Summe ergibt sich für mich, dass ein Wechsel von solarprognose auf solcast keinen nennenswerten Vorteil bringt. In Summe ist solarprognose minimal "näher dran" an der Realität, wobei beide Dienste auch mal Ausreißer haben.

Ach so, bei solarprognose arbeite ich mit "mosmix" als Quelle.

Gruß,
JD
